sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/SonarAnalysisContext.cs

```diff
@@ -70,6 +72,21 @@ public class SonarAnalysisContext
         /// </remarks>
         public static Action<IReportingContext> ReportDiagnostic { get; set; }
 
+        public static bool ShouldAnalyzeGenerated(AnalysisContext analysisContext, Compilation c, AnalyzerOptions options) =>
+            TryGetSonarLintXml(options, out var sonarLintXml) &&
+            analysisContext.TryGetValue(sonarLintXml.GetText(), GetProvider(c.Language), out var shouldAnalyzeGeneratedCode) &&
+            shouldAnalyzeGeneratedCode;
+
+        public static bool ShouldAnalyzeGenerated(CompilationStartAnalysisContext analysisContext, Compilation c, AnalyzerOptions options) =>
+            TryGetSonarLintXml(options, out var sonarLintXml) &&
+            analysisContext.TryGetValue(sonarLintXml.GetText(), GetProvider(c.Language), out var shouldAnalyzeGeneratedCode) &&
+            shouldAnalyzeGeneratedCode;
+
+        public static bool ShouldAnalyzeGenerated(CompilationAnalysisContext analysisContext, Compilation c, AnalyzerOptions options) =>
+            TryGetSonarLintXml(options, out var sonarLintXml) &&
+            analysisContext.TryGetValue(sonarLintXml.GetText(), GetProvider(c.Language), out var shouldAnalyzeGeneratedCode) &&
+            shouldAnalyzeGeneratedCode;

```

Comment:
We need tests to check that the caching actually works.
I've taken the liberty of adding some in another commit that I've pushed to this branch.

Reply:
You, dear Sir, are a true gentleman!

Reply:
You're welcome :-)